src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -703,10 +759,10 @@ private static string GetPropertyName(MemberInfo property!!)
                     .Value?
                     .ToString();
 
-                return !string.IsNullOrWhiteSpace(name) ? name : property.Name;
```

Comment:
We could add this support in a separate PR. The idea would be to allow usage of `ConfigurationKeyName` attribute over on constructor parameters, not just over a property as it is defined today [here](https://source.dot.net/#Microsoft.Extensions.Configuration.Abstractions/ConfigurationKeyNameAttribute.cs,8).

#### Sample usage:

With records:

```c#
public record Poco([ConfigurationKeyName="my_string"] string myString);
```

Workaround without this would be to add the annotation over the property as shown below:

```c#
public class Poco
{
    [ConfigurationKeyName="my_string"]
    public string MyString { get; init; }

    public Poco(string myString) { .. }
}
```

TODO: 

- [ ] Add API proposal for this change (outside the scope for this PR).

Reply:
Thanks @maryamariyan - is the task for me, or for somebody else?

Reply:
I will take this task to API review.

Reply:
Shall I create a branch off this branch and implement it?

Reply:
We'd be able to add that only after the API review gets approved.

Reply:
OK. I think I've addressed all the issues in this PR then. Happy to respond to any more.